File: TransCoder_tokenized_test_set_functions/java/CHECK_NUMBER_POWER_K_USING_BASE_CHANGING_METHOD.java
static boolean isPowerOfK ( int n, int k ) {
  boolean oneSeen = false;
  while ( n > 0 ) {
    int digit = n % k;
    if ( digit > 1 ) return false;
    if ( digit == 1 ) {
      if ( oneSeen ) return false;
      oneSeen = true;
    }
    n /= k;
  }
  return true;
}

File: TransCoder_tokenized_test_set_functions/python/CHECK_NUMBER_POWER_K_USING_BASE_CHANGING_METHOD.py
def isPowerOfK ( n, k ) :
    oneSeen = False
    while ( n > 0 ) :
        digit = n % k
        if ( digit > 1 ) :
            return False
        if ( digit == 1 ) :
            if ( oneSeen ) :
                return False
            oneSeen = True
        n //=k
    returnTrue
